File: src/main/java/com/github/nsc/de/compiler/parser/node/variables/VariableSubAssignmentNode.java
package com.github.nsc.de.compiler.parser.node.variables;

import com.github.nsc.de.compiler.parser.node.Node;
import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class VariableSubAssignmentNode implements ValuedNode {

    private final ValuedNode variable;
    private final Node value;

    public VariableSubAssignmentNode(ValuedNode variable, Node value) {
        this.variable = variable;
        this.value = value;
    }

    public ValuedNode getVariable() {
		return variable;
	}

    public Node getValue() {
        return value;
    }

    @Override
    public String toString() {
        return '{' + getValue().toString() + "-=" + this.getValue().toString() + '}';
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/variables/VariableSubAssignmentNode.kt
package com.github.nsc.de.shake.parser.node.variables

import com.github.nsc.de.shake.parser.node.*
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class VariableSubAssignmentNode(
    map: PositionMap,
    val variable: ValuedNode,
    val value: Node,
    val operatorPosition: Int
) : ValuedNode(map) {

    override fun toJson(): Map<String, *> = mapOf(
        "name" to "VariableSubAssignmentNode",
        "variable" to variable.json,
        "value" to value.json
    )

}